doc/getting-started/Kotlin-Versions.md

```diff
@@ -0,0 +1,15 @@
+## Kotlin versions and the Gradle Kotlin DSL 
+
+Gradle ships with Kotlin DSL support which uses embedded Kotlin compiler, plus fixed Kotlin `stdlib` and `reflect` dependencies. For example, Gradle 4.3 ships with Kotlin 1.1.51, as well as `stdlib` and `reflect` with the same version. The `kotlin` package from those modules is visible through the Gradle classpath.
+
+For both backward and forward compatibility, the [compatibility guarantees](https://kotlinlang.org/docs/reference/compatibility.html) provided by Kotlin apply.
+
+### Backward compatibility
+
+Our approach is to only do backwards-breaking Kotlin upgrades on a major Gradle release. We will always clearly document which Kotlin version we ship and announce upgrade plans before a major release. 
+
+Plugin authors who want to stay compatible with older Gradle versions need to limit their API usage to a subset that is compatible with these old versions. It’s not really different from any other new API in Gradle. E.g. if we introduce a new API for dependency resolution and a plugin wants to use that API, then they either need to drop support for older Gradle versions or they need to do some clever organization of their code to only execute the new code path on newer versions.
```

Comment:
How will API versioning be indicated on various methods/files? In the kotlin docs?
There's a library called [API Guardian](https://github.com/apiguardian-team/apiguardian) (created for and used by the Junit 5 team) that may work for this.

Reply:
Let's move this as a follow up. Could you please create an issue for it?

Reply:
Sure.

Reply:
See #534